ContestModel/src/org/icpc/tools/contest/model/internal/Contest.java

```diff
@@ -146,6 +151,19 @@ public void add(IContestObject obj) {
 			lastTimedEvent = obj;
 			lastTimedEventIndex = data.size();
 		}
+
+		// update known properties
+		int ord = obj.getType().ordinal();
+		List<String> knownProps = allKnownProperties[ord];
+		if (knownProps == null) {
+			knownProps = new ArrayList<String>();
```

Comment:
Set is definitely the right external interface too, but switching to HashSet is likely worse performance (small sets, and HashSet sits on top of a HashMap) and isn't ordered, which meant the properties appear random. I know this doesn't really matter, but I used overkill and made a SimpleSet from SimpleMap. 'id' is always the first property again and others follow the same logical order as json, performance is as good as it can get.